crates/shared/src/token_list.rs

```diff
@@ -62,6 +92,14 @@ struct TokenModel {
     token: Token,
 }
 
+#[async_trait::async_trait]
+impl Maintaining for TokenList {
```

Comment:
At some point it might make sense to add an abstraction that doesn't necessarily run on every maintenance loop (but instead has a timer and noops until the desired time is reached). This should be it's own implementation of `Maintaining` that just defers into the concrete e.g. `TokenList` impl

Reply:
Or maybe extend the `Maintaining` abstraction to specify _when_ it wants to run.
```rs
pub enum MaintenanceCondition {
  NewBlock,
  Timer(Duration),
}
```

At that point I'd refactor the `Maintaining` abstraction into something with a more general name such as `BackgroundTask` or `BackgroundLooping`.

Reply:
On second thought I don't think that a background task is the same thing as a block listener. Seems like it's mixing two entirely different abstractions.

Reply:
Makes sense to avoid update on each new block, especially for chains with lower block times.
Ended up using the most simple solution - adding cache with `updated_at` field. All other options seem to me like out of scope of this PR.

Reply:
The maintenance abstraction was explicitly meant for updates that happen on each block. If this component doesn't do that then I wouldn't include it here. I think we should manually tokio::spawn this new updating task. Not sure if an overarching abstraction is needed.

Reply:
I see, I guess one could argue that source of truth (ENS record) could update on every block (however we don't really want to load this list every 12s). 
You guys know better if this should be part of per block maintenance or spawned in a different task. Also if this is blocking/bloating this PR or should be done in a follow up.

Reply:
Using maintenance on each block seems a bit like misusing it's purpose. 
Extracted into separate background task.